WordPress/gutenberg
# Improve the parsing of remote url details in the URL Details endpoint to REST API

Builds on the original implementation from https://github.com/WordPress/gutenberg/pull/18042 to add a better mechanism for parsing the information from the remote URL's response body HTML.

This helps to address feedback such as https://github.com/WordPress/gutenberg/issues/27762 and also sets things up nicely for parsing additional detail from the remote URL (eg: favicon, Open Graph meta...etc).

This PR focuses solely on utilising `DOMDocument` and friends to improve the parsing mechanic.

## Automated Testing Instructions

`npm run test-php` should run the unit tests which reside in `phpunit/class-wp-rest-url-details-controller-test.php`.

## Manual Testing Instructions

* Check out this PR.
* Run `npm run wp-env start` to boot the local testing env.
* Open the Gutenberg local testing env at `http://localhost:8889/wp-admin/` and login.
* Create a new Post.
* Choose one of the following options:

### Option 1

* Open devtools and in the `console` enter:

```
wp.apiFetch( { path: '/__experimental/url-details/?url=https://wordpress.org' } ).then( data => {
    console.log( data );
} );
```

...also try some non-English websites:

```
wp.apiFetch( { path: '/__experimental/url-details/?url=http://www.baidu.com' } ).then( data => {
    console.log( data );
} );
```


You'll see the response returned there.

### Option 2
* Open devtools and in the `console` enter `wpApiSettings.nonce`. You should see a valid nonce returned as a string.
* Copy the nonce.
* In your browser goto: `http://localhost:8889/wp-json/__experimental/url-details/?url=https://wordpress.org&_wpnonce=%YOUR_NONCE_HERE%` - be sure to replace the nonce value with that copied from the previous step.
* You should see a valid REST API response containing the contents of the title tag from `wordpress.org` (ie: `"Blog Tool, Publishing Platform, and CMS \u2014 WordPress.org"`).


## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
